libopae/src/enum.c

```diff
@@ -547,6 +567,22 @@ enum_top_dev(const char *sysfspath, struct dev_list *list)
 
 	pdev->bus = (uint8_t) b;
 
+	// read the vendor and device ID from the 'device' path
+	uint32_t x = 0;
+	char vendorpath[SYSFS_PATH_MAX];
+	snprintf_s_s(vendorpath, SYSFS_PATH_MAX, "%s/device/vendor", sysfspath);
+	result = sysfs_read_u32(vendorpath, &x);
+	if (result != FPGA_OK)
+		return result;
+	pdev->vendor_id = (uint16_t)x;
```

Comment:
Casting to a narrower int type may upset the compiler, and probably klockwork?
Same comment applies to other similar casts in this PR.